### PR TITLE
Sync rightnow

### PR DESCRIPTION
**Pull request by brong**
This branch is now hopefully ready to run.  It does real time replication of many things, and still works with fallback as well.  It won't do anything until we deploy the config changes to turn it on.

### CODE REVIEW

File: imap/http_dav.c [hunk 1]
```diff
@@ -6409,6 +6421,9 @@ EXPORTED int meth_propfind(struct transaction_t *txn, void *params)
     xml_partial_response(txn, fctx.root->doc, NULL /* end */, 0, &fctx.xmlbuf);
     xmlBufferFree(fctx.xmlbuf);
 
+    // might have made a change!
+    sync_checkpoint(txn->conn->pin);
+
```
Comment (ksmurchison) [inline imap/http_dav.c:6426]: @brong maybe its just too early for me, but what could PROPFIND actually change.  Its a safe (read-only) method

File: imap/http_dav.c [hunk 2]
```diff
@@ -8244,6 +8265,9 @@ int meth_report(struct transaction_t *txn, void *params)
     /* Process the requested report */
     if (!ret) ret = (*report->proc)(txn, rparams, inroot, &fctx);
 
+    // might have made a change!
+    sync_checkpoint(txn->conn->pin);
+
```
Comment (ksmurchison) [inline imap/http_dav.c:8270]: @brong Same question as for PROPFIND

**Review by ksmurchison — APPROVED** [review]
Other than my 2 inline questions, this looks good to me.

File: imap/http_dav.c (hunk 1 above)
Comment (brong) [reply imap/http_dav.c:6426]: It can create mailboxes if they don't exist yet!

File: imap/http_dav.c (hunk 2 above)
Comment (brong) [reply imap/http_dav.c:8270]: same answer :P

**Comment by brong** [comment]
@ksmurchison if no changes are made, sync_checkpoint is a noop anyway, because the log is empty :)

**Pull request merged**